src/middleware/messageStore.js

```diff
@@ -174,19 +190,49 @@ export async function storeResponse (ctx, done) {
     update.orchestrations.push(...ctx.orchestrations)
   }
 
-  await extractTransactionPayloadIntoChunks(update)
+  //await extractTransactionPayloadIntoChunks(update)
 
-  return transactions.TransactionModel.findOneAndUpdate({_id: ctx.transactionId}, update, {runValidators: true}, (err, tx) => {
+  transactions.TransactionModel.findOneAndUpdate(transactionId, update, { runValidators: true }, (err, tx) => {
     if (err) {
-      logger.error(`Could not save response metadata for transaction: ${ctx.transactionId}. ${err}`)
-      return done(err)
+      logger.error(`Could not save initial response metadata for transaction: ${ctx.transactionId}. ${err}`)
+      done(err)
+    }
+    if ((tx === undefined) || (tx === null)) {
+      logger.error(`Could not find transaction: ${ctx.transactionId}`)
+      done(err)
+    }
+    logger.info(`stored initial primary response for ${tx._id}`)
+    done(null, tx)
+  })
+}
+
+/*
+ *  Find and update an existing transaction once a Response has completed streaming 
+ *    into the HIM (Not async; Mongo should handle locking issues, etc)
+ */
+export function completeResponse (ctx, done) {
+
+  if (ctx && !ctx.responseTimestampEnd) {
+    ctx.responseTimestampEnd = new Date()
+  }
+ 
+  const transactionId = getTransactionId(ctx)
+
+  const update = {
+    'response.timestampEnd': ctx.responseTimestampEnd
+  }
+  
+  return transactions.TransactionModel.findOneAndUpdate(transactionId, update, {runValidators: true}, (err, tx) => {
+    if (err) { 
+      logger.error(`Could not save completed response metadata for transaction: ${ctx.transactionId}. ${err}`)
+      return done(err) 
     }
     if ((tx === undefined) || (tx === null)) {
       logger.error(`Could not find transaction: ${ctx.transactionId}`)
       return done(err)
     }
-    logger.info(`stored primary response for ${tx._id}`)
-    return done()
+    logger.info(`stored completed primary response for ${tx._id}`)
```

Comment:
I think we can remove the text `completed` as this might be interpreted as the transaction status (Successful, Completed, Failed etc)

Reply:
addressed

Reply:
commit fixing this - https://github.com/jembi/openhim-core-js/pull/1036/commits/7afbbafb09eb82730888261e1ef66c9228120fc6